filesmanager/tests/test_filesmanager.py

```diff
@@ -0,0 +1,949 @@
+        Expected result:
+            - The method returns the correct content, index, and parent directory.
+        """
+        name = "Unpublished"
+        parent_content = self.xblock.directories["children"]
+
+        expected_result = (parent_content[0], 0, parent_content)
+
+        result = self.xblock.get_content_by_name(name, parent_content)
+
+        self.assertEqual(expected_result, result)
+
+    def test_get_content_by_name_content_does_not_exist(self):
+        """
+        Check get_content_by_name method when content does not exist.
+
+        Expected result:
+            - The method returns (None, None, None).
+        """
+        name = "Nonexistent"
+        parent_content = self.xblock.directories["children"]
+
+        expected_result = (None, None, None)
+
+        result = self.xblock.get_content_by_name(name, parent_content)
+
+        self.assertEqual(expected_result, result)
+
+    def test_get_content_by_name_content_in_subdirectory(self):
+        """
+        Check get_content_by_name method when content is in a subdirectory.
+
+        Expected result:
+            - The method returns the correct content, index, and parent directory.
+        """
+        name = "SubdirectoryContent"
+        parent_content = self.xblock.directories["children"]
+        parent_content[0]["children"].append(
+            {
+                "id": "subdirectorycontent",
+                "name": "SubdirectoryContent",
+                "type": "file",
+                "path": "Root/Unpublished/SubdirectoryContent",
+                "metadata": {},
+                "parentId": "unpublished",
+            }
+        )
+
+        expected_result = (
+            parent_content[0]["children"][0],
+            0,
+            parent_content[0]["children"],
+        )
+
+        result = self.xblock.get_content_by_name(name, parent_content)
+
+        self.assertEqual(expected_result, result)
+
+    def test_get_content_by_path_root(self):
+        """
+        Check get_content_by_path method when path is "Root".
+
+        Expected result:
+            - The method returns the root directory, None, None.
+        """
+        path = "Root"
+
+        expected_result = (self.xblock.directories, None, None)
+
+        result = self.xblock.get_content_by_path(path)
+
+        self.assertEqual(expected_result, result)
+
+    def test_get_content_by_path_content_exists(self):
+        """
+        Check get_content_by_path method when content exists.
+
+        Expected result:
+            - The method returns the correct content, index, and parent directory.
+        """
+        path = "Root/Unpublished"
+
+        expected_result = (
+            self.xblock.directories["children"][0],
+            0,
+            self.xblock.directories["children"],
+        )
+
+        result = self.xblock.get_content_by_path(path)
+
+        self.assertEqual(expected_result, result)
+
+    def test_get_content_by_path_content_does_not_exist(self):
```

Comment:
Thank you so much for all these test cases! I recommend you check out the ddt library. This way we could cover multiple inputs with a single test case. Here are some examples:
https://github.com/eduNEXT/xblock-limesurvey/blob/main/limesurvey/tests/test_limesurvey.py#L348-L352
You don't need to do it for this PR, but you already know next time. Thanks again!